### ISSUE TITLE
Colorization issues

### ISSUE THREAD

**Issue by tpetricek** [fsharp/FSharpMDXS#6]
I probably don't have the latest version, so this might already be fixed, but I keep running into issues caused by the fact that F# compiler tokenization is more sophisticated than the regex (or whatever)-based grammar files used by Xamarin.

Some simple things that break are:

- Using triple `"` comments:

        let foo = """ hello " world """

- Using nested comments:

        let foo (* (* nested *) *) = 1

There are a few more things - but I guess it really boils down to the limitations of the current colorization implementation. Are there any plans for using F# compiler service for this?

BTW: I also get intellisense in comments and strings, which is not very useful :-)

**Comment by 7sharp9** [comment]
`let foo = """ hello " world """` and `let foo (* (* nested *) *) = 1` both work in 5.9.

The was highlighting works is that the regex based xml definition is applied to the initial text, via span parsing etc.  When a compilation parse is completed asynchronously we use a combination of the lexer and ast to colourise.  I am interested in optimising this as there can sometimes be a lag, especially if your editing code in the compiler where compilation can take many seconds on a single large file.

**Comment by 7sharp9** [comment]
![screen shot 2015-03-26 at 10 41 15](https://cloud.githubusercontent.com/assets/588746/6844630/af69ed90-d3a4-11e4-974e-1d980a7b4489.png)

**Comment by tpetricek** [comment]
It looks like the most obvious issues are now fixed, which is great :-).

Though, I still found a case where it does not quite work:
![screen shot 2015-04-24 at 00 03 08](https://cloud.githubusercontent.com/assets/485413/7309363/8a55d4c4-ea15-11e4-9e2d-2c5c8efd2abb.png)

If you type `let a = '"'` into F# interactive, it works just fine.
Another issue with the same block of code was that the colorization breaks (and recovers only after I re-open the file) when I type something into the comment:
![screen shot 2015-04-24 at 00 03 21](https://cloud.githubusercontent.com/assets/485413/7309384/bae4587c-ea15-11e4-8250-cd43be0d1708.png)

I have Xamarin Studio 5.8.3 (build 1) and F# Language Binding Version 5.8.0 (although the comment says it is "for Xamarin Studio/MonoDevelop 5.8.3").

Let me know if you cannot repro this and I'll share the full project where this happens.

**Comment by 7sharp9** [comment]
Looks like this works ok in 5.9:

![screen shot 2015-04-24 at 09 54 40](https://cloud.githubusercontent.com/assets/588746/7315731/219d653e-ea68-11e4-984e-a2718f60b949.png)

**Comment by tpetricek** [comment]
Neat, I'll try getting the beta version of Xamarin Studio (assuming that's all I need?)

**Comment by 7sharp9** [comment]
5.9. is now in stable too.
